met-web/src/components/engagement/listing/index.tsx

```diff
@@ -28,10 +26,47 @@ import { USER_ROLES } from 'services/userService/constants';
 import { ApprovedIcon, NewIcon, NFRIcon, RejectedIcon } from './Icons';
 import { CommentStatus } from 'constants/commentStatus';
 import { ActionsDropDown } from './ActionsDropDown';
-import { updateURLWithPagination } from 'components/common/Table/utils';
 import AdvancedSearch from './AdvancedSearch/SearchComponent';
 import { Button, TextInput } from 'components/common/Input';
 import { faPlus } from '@fortawesome/pro-regular-svg-icons';
+import { AutoBreadcrumbs } from 'components/common/Navigation/Breadcrumb';
+
+interface SearchFilter {
+    key?: string;
+    value?: string;
+}
+
+// Create a search, filter, and pagination string to pass values to the route.
+const updateURL = (
+    searchFilter: SearchFilter,
+    paginationOptions: PaginationOptions<Engagement>,
+    searchOptions: SearchOptions,
+) => {
+    const statusList: string = searchOptions.status_list.join('_').toString();
+    const url = new URL(window.location.href);
+    searchFilter?.value
+        ? url.searchParams.set('search_text', searchFilter.value)
+        : url.searchParams.delete('search_text');
+    paginationOptions?.page && url.searchParams.set('page', paginationOptions.page.toString());
+    paginationOptions?.size && url.searchParams.set('size', paginationOptions.size.toString());
+    paginationOptions?.sort_key &&
+        url.searchParams.set('sort_key', 'engagement.' + paginationOptions.sort_key.toString());
```

Comment:
nested_sort_key could probably be used here instead since it includes the `model.` prefix

Reply:
Hm ok, I tried printing out the nested sort key value at first and it seems to function differently from the sort_key. Is this not a secondary sort?